### PR TITLE
Testing for exposed admin urls

### PR DESCRIPTION
**Pull request by willbarton**
We've been talking about admin URLs, and ensuring that we don't expose anything we don't want to expose. We have an `ALLOW_ADMIN_URL` setting that we use to turn admin URLs on and off on a per-Django-instance basis, but there may be situations where we want to enable admin URLs internally but not externally for a single Django instance. Those URLs might be blacklisted between the Django instance and the public, however. For this reason, we need to verify that we always know exactly which admin URLs are enabled by `ALLOW_ADMIN_URL`. This PR adds a test that introspects our URLs and verifies that we're not exposing anything that we don't know about.

This test will work with our optional apps when they're included by using `tox -e optional`. That is where the real benefit of this test comes from — satellite apps could add URLs that use `ALLOW_ADMIN_URL` and we might not be aware of it otherwise.
 
## Additions

- Test for `ALLOW_ADMIN_URL` urls 

## Checklist

* [x] PR has an informative and human-readable title
* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [x] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: cfgov/cfgov/tests/test_urls.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+try:
```
Comment (richaagarwal) [inline cfgov/cfgov/tests/test_urls.py]: When/why would this fail?

File: cfgov/cfgov/tests/test_urls.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    urls = []
+    for p in urlpatterns:
+        if isinstance(p, RegexURLPattern) or hasattr(p, '_get_callback'):
+            urls.append(base + p.regex.pattern)
+        elif (isinstance(p, RegexURLResolver) or
+              hasattr(p, 'url_patterns') or
+              hasattr(p, '_get_url_patterns')):
+            try:
+                patterns = p.url_patterns
+            except ImportError:
```
Comment (richaagarwal) [inline cfgov/cfgov/tests/test_urls.py]: Why would this throw an `ImportError` ?

File: cfgov/cfgov/tests/test_urls.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    '^logout',
+    '^tasks/',
+    '^csp-report/',
+    '^password/',
+    '^d/admin/',
+]
+
+
+# Based on django_extensions's show_urls command.
+def get_urlpatterns(urlpatterns, base=''):
```
Comment (richaagarwal) [inline cfgov/cfgov/tests/test_urls.py]: Can you add a docstring here? It's not clear what this function is doing

File: cfgov/cfgov/tests/test_urls.py (hunk 1 above)
Comment (willbarton) [inline cfgov/cfgov/tests/test_urls.py]: This is for Django 1.10+ compatibility. `RegexURLPattern` and `RegexURLResolver` moved to `django.urls` in 1.10.

File: cfgov/cfgov/tests/test_urls.py (hunk 2 above)
Comment (willbarton) [inline cfgov/cfgov/tests/test_urls.py]: I'm not sure — the code this is based on catches an `ImportError` there. Removing it doesn't throw an `ImportError` with cfgov-refresh or any of its optional apps. Removing it.

**Comment by willbarton** [comment]
@richaagarwal good comments on late-night-written code. I've renamed that function to a much more descriptive `extract_regexes_from_urlpatterns`, because that's what it actually does.

File: cfgov/cfgov/tests/test_urls.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    regexes = []
+    for p in urlpatterns:
+        if isinstance(p, RegexURLPattern) or hasattr(p, '_get_callback'):
+            regexes.append(base + p.regex.pattern)
+        elif (isinstance(p, RegexURLResolver) or
+              hasattr(p, 'url_patterns') or
+              hasattr(p, '_get_url_patterns')):
+            patterns = p.url_patterns
+            regexes.extend(extract_regexes_from_urlpatterns(
+                p atterns, base + p.regex.pattern))
```
Comment (richaagarwal) [inline cfgov/cfgov/tests/test_urls.py]: typo here causing test to fail

**Pull request merged**